### PR TITLE
Added quantization options

### PR DESCRIPTION
**Pull request by javafa**
Added quantization options
when use quantization, selected_indexs in compute_logps() could be negative values

### CODE REVIEW

**Comment by javafa** [comment]
loss calculation is wrong.

**Comment by jiwooya1000** [comment]
Thank you @javafa for adding the quantization options! I will check it soon and merge it if it is good to go👍

**Comment by javafa** [comment]
calculation nagative values should be changed
...
selected_index[selected_index < 0] += filtered_logits.size(1)
...
